server/src/main/java/org/apache/druid/segment/realtime/firehose/EventReceiverFirehoseFactory.java

```diff
@@ -79,6 +79,7 @@
  * Builds firehoses that accept events through the {@link EventReceiver} interface. Can also register these
  * firehoses with an {@link ServiceAnnouncingChatHandlerProvider}.
  */
+@Deprecated
```

Comment:
I think `TimedShutoffFirehoseFactory` and `ClippedFirehoseFactory` need marked deprecated too?

Reply:
Just wanted to know, since `FirehoseFactory` is marked deprecated, does it imply the implementations are deprecated as well? In that case, me marking `EventReceiverFirehoseFactory` as deprecated already is redundant and will remove it, in otherwise scenario I'll mark all 4 deprecated.

Reply:
I think it doesn't hurt to mark them all with `@Deprectated` since it doesn't look like intellij visually marks them as deprecated (with the strikethrough) if just the interface is deprecated.